content/docs/1.5.1/volumes-and-nodes/internal-data-structures.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Internal Data Structures
+weight: 3
+---
+
+**Table of Contents**
+- [Node Disk](#node-disk)
+  - [Filesystem-Type Disk](#filesystem-type-disk)
+  - [Block-Type Disk](#block-type-disk)
+- [Replica Directory on Node Disk](#replica-directory-on-node-disk)
+
+## Node Disk
+
+### Filesystem-Type Disk
+
+A filesystem-type disk contains a configuration file named `longhorn-disk.cfg` and a directory called `replicas` that stores replica directories for volumes. The `longhorn-disk.cfg` file contains a JSON object with a single key `diskUUID` whose value is a 36-character UUID string, as shown below:
+```
+{
+  "diskUUID": "${UUID}"
+}
+```
+
+### Block-Type Disk
+
+A block-type disk is managed by the Storage Performance Development Kit (SPDK). SPDK constructs a logical volume store (lvstore) that utilizes the super blob feature of the blobstore to hold the UUID and other metadata. The blobstore types are implemented within the blobstore itself and are persisted on disk.
+
+## Replica Directory on Node Disk
+
+When creating a Longhorn V1 volume, the Longhorn system creates its downstream replicas on filesystem-type disks. The path of a replica directory follows the pattern ${DISK_PATH}/replicas/${VOLUME_NAME}-${8-CHAR_ID}. A newly created volume replica directory will contain the following files:
+
+```
+revision.counter
+volume-head-000.img
+volume-head-000.img.meta
+volume.meta
+```
+
+- revision.counter: Records the latest revision count of the replica for the volume.
+- volume-head-000.img: The actual data file containing the volume data.
+- volume-head-000.img.meta: Stores metadata related to the volume-head-000.img file, including the file name, parent file, creation timestamp, and other attributes.
+- volume.meta: Records the attributes of the volume, such as size, current head file, parent snapshot, and other metadata.
+
+After creating multiple snapshots, the disk chain will look like this:
+
+```
+snap 1 -----> snap 2 -----> ... -----> snap N -----> volume head
+```
+
+The file list of a practical volume containing two snapshots and one volume head might look like this:
+
+```
+revision.counter
+volume-head-002.img
+volume-head-002.img.meta
+volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img
+volume-snap-2e8bf0e9-3751-479c-a560-96a275d2672c.img.meta
+volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img
+volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img.meta
+volume.meta
+```
+
+In this example, the disk chain is:
+```
+96a275d2672c -----> 9a61c5ff0907 -----> volume-head-002
+```
+Where 96a275d2672c and 9a61c5ff0907 are abbreviations for the snapshot file names.
+
+The contents of the volume.meta and image metadata files (*.img.meta) are:
+```
+// volume.meta
+{
+  "Size": 2147483648,
+  "Head": "volume-head-002.img",
+  "Dirty": true,
+  "Rebuilding": false,
+  "Error": "",
+  "Parent": "volume-snap-dd51b6b5-c62a-4676-a0d1-9a61c5ff0907.img",
+  "SectorSize": 512,
+  "BackingFilePath": ""
+}
```

Comment:
I have a request, though I think it's reasonable for you to disagree about whether this fits the doc.

As a developer, the meaning of a few things is not immediately clear until I go digging. Since were documenting these internal structures, can we clarify somewhere in this doc:

- What is meant by `dirty`?
- Why does the `volume-head` name include an incrementing integer?